include/boost/geometry/algorithms/detail/buffer/line_line_intersection.hpp

```diff
@@ -24,23 +24,97 @@ namespace boost { namespace geometry
-        // their join type is checked before.
-        return arithmetic::intersection_point(p, q, ip);
+    template <typename Point>
+    static Point between_point(Point const& a, Point const& b)
+    {
+        Point result;
+        geometry::set<0>(result, (geometry::get<0>(a) + geometry::get<0>(b)) / 2.0);
+        geometry::set<1>(result, (geometry::get<1>(a) + geometry::get<1>(b)) / 2.0);
+        return result;
+    }
+
+    template <typename Point>
+    static bool
+    apply(Point const& pi, Point const& pj, Point const& qi, Point const& qj,
+          Point const& vertex, bool equidistance, Point& ip)
+    {
+        // Calculates ip (below) by either intersecting p (pi, pj)
+        // with q (qi, qj) or by taking a point between pj and qi (b) and
+        // intersecting r (b, v), where v is the original vertex, with p (or q).
+        // The reason for dual approach: p might be nearly collinear with q,
+        // and in that case the intersection points can lose precision
+        // (or be plainly wrong).
+        // Therefore it takes the most precise option (this is usually p, r)
+        //
+        //             /qj                     |
+        //            /                        |
+        //           /      /                  |
+        //          /      /                   |
+        //         /      /                    |
+        //        /qi    /                     |
+        //              /                      |
+        //   ip *  + b * v                     |
+        //              \                      |
+        //        \pj    \                     |
+        //         \      \                    |
+        //          \      \                   |
+        //           \      \                  |
+        //            \pi    \                 |
+        //
+        // If generated sides along the segments can have an adapted distance,
+        // in a custom strategy, then the calculation of the point in between
+        // might be incorrect and whe optimization is not used.
+
+        using ct = typename coordinate_type<Point>::type;
+
+        auto const p = detail::make::make_infinite_line<ct>(pi, pj);
+        auto const q = detail::make::make_infinite_line<ct>(qi, qj);
+        auto const denominator_pq = p.b * q.a - p.a * q.b;
+        constexpr decltype(denominator_pq) const zero = 0;
+
+        if (equidistance)
+        {
+            auto const between = between_point(pj, qi);
+
+            auto const r = detail::make::make_infinite_line<ct>(vertex, between);
+
+            auto const denominator_pr = p.b * r.a - p.a * r.b;
+
+            if (math::equals(denominator_pq, zero)
+                && math::equals(denominator_pr, zero))
+            {
+                // Degenerate case (for example when length results in <inf>)
+                return false;
+            }
+
+            ip = std::abs(denominator_pq) > std::abs(denominator_pr)
```

Comment:
This is the fix

Reply:
So are we using `std::abs` now instead of `math::abs`?

Reply:
Thanks, that was an error. Fixed. :heavy_check_mark: